From e254d50fbe3ce2d70458256c724b62bfc13aa5e2 Mon Sep 17 00:00:00 2001
From: Alan Ott <alan@signal11.us>
Date: Fri, 29 Mar 2013 16:49:27 -0400
Subject: [PATCH 147/177] firmware: capes: Update MRF24J40 cape to work with
 latest bone dtsi

am335x-bone-common.dtsi has removed the pinmux settings for SPI1.
Add them to the MRF24J40 cape dts fragment

Signed-off-by: Alan Ott <alan@signal11.us>
---
 firmware/capes/cape-bone-mrf24j40-00A0.dts | 25 ++++++++++---------------
 1 file changed, 10 insertions(+), 15 deletions(-)

diff --git a/firmware/capes/cape-bone-mrf24j40-00A0.dts b/firmware/capes/cape-bone-mrf24j40-00A0.dts
index cdef7e2..0fc2aea 100644
--- a/firmware/capes/cape-bone-mrf24j40-00A0.dts
+++ b/firmware/capes/cape-bone-mrf24j40-00A0.dts
@@ -23,19 +23,22 @@
 				pinctrl-single,pins = <
 					/* Pinmux comes from TRM section
 					   9.3.1: CONTROL_MODULE Regusters. */
-					//0x44 0x2f /* gpmc_a1.gpio1_17 INPUT,no-pull,mode7 */
-					//0x1A4 0x2f /* mcasp0_fsr.gpio3_19 INPUT,no-pull,mode7 */
-					//0x00 0x2f /* gpmc_ad0.gpio1_0 */
 
-					//so use GPIO1_13 (rst), GPIO1_14 (wake), then SPI1_CS0 (CS), and GPIO1_29 (INT)
+					/* so use GPIO1_13 (rst), GPIO1_14 (wake), then SPI1_CS0 (CS), and GPIO1_29 (INT) */
 					0x34 0x37 /* gpmc_ad13.gpio1_13 input, pull up, mode 7 */
 					0x38 0x2f /* gpmc_ad14.gpio1_14 input, no pull, mode 7 */
-					//0x13c 0x2f /* gpmc_ad15.gpio1_15 input, no pull, mode 7 */
 					0x7c 0x2f /* gpmc_csn0.gpio1_29 input, no pull, mode 7 */
 				>;
 			};
+			bone_mrf24j40_spi1_pins: pinmux_bone_mrf24j40_spi1_pins {
+				pinctrl-single,pins = <
+					0x190 0x33	/* mcasp0_aclkx.spi1_sclk, RX_ENABLED | PULLUP | MODE3 */
+	 				0x194 0x33	/* mcasp0_fsx.spi1_d0, RX_ENABLED | PULLUP | MODE3 */
+					0x198 0x33	/* mcasp0_axr0.spi1_d1, RX_ENABLED | PULLUP | MODE3 */
+					0x19c 0x33	/* mcasp0_ahclkr.spi1_cs0, RX_ENABLED | PULLUP | MODE3 */
+				>;
+			};
 		};
-
 	};
 
 
@@ -49,7 +52,7 @@
 
 			status = "okay";
 			pinctrl-names = "default";
-			pinctrl-0 = <&spi1_pins_s0>;
+			pinctrl-0 = <&bone_mrf24j40_spi1_pins>;
 
 			mrf24j40@0 {
 				compatible = "mrf24j40ma";
@@ -62,14 +65,6 @@
 				interrupt-parent = <&gpio2>;
 				interrupts = <29>;
 
-				// also working feb 9
-				//interrupt-parent = <&gpio2>;
-				//interrupts = <0>;
-
-				//working configuration
-				//interrupt-parent = <&gpio2>;
-				//interrupts = <17>;
-
 				//mrf24j40-wake = <&gpio4 19>;
 				//mrf24j40-reset = <gpio21 0>;
 			};
-- 
1.8.1.4

